packages/shorebird_cli/lib/src/commands/releases/list_releases_command.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:collection/collection.dart';
+import 'package:intl/intl.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/commands/preview_command.dart';
+import 'package:shorebird_cli/src/config/config.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/shorebird_command.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template list_releases_command}
+/// A command to list available releases.
+/// {@endtemplate}
+class ListReleasesCommand extends ShorebirdCommand {
+  /// {@macro list_releases_command}
+  ListReleasesCommand() {
+    argParser
+      ..addOption(
+        'flavor',
+        help: 'The flavor for which to list releases.',
+      )
+      ..addOption(
+        'limit',
+        help: 'Limit number of releases to be printed.',
+        defaultsTo: '$defaultLimit',
+      );
+  }
+
+  /// The default limit for the number of releases to be printed.
+  static const int defaultLimit = 10;
+
+  @override
+  String get description => 'List available releases.';
+
+  @override
+  String get name => 'list';
+
+  /// The shorebird app ID for the current project.
+  String get appId => shorebirdEnv.getShorebirdYaml()!.getAppId(flavor: flavor);
+
+  /// The build flavor, if provided.
+  late String? flavor = results['flavor'] as String?;
+
+  /// Whether to only show the latest release for each platform.
+  late int limit = int.tryParse(
+        results['limit'] as String? ?? '$defaultLimit',
+      ) ??
+      defaultLimit;
```

Comment:
It's a little gnarly to have the limit parsing code split between this property and the `> 0` validation that happens in the `run` method. I don't feel strongly which you keep, but it should probably exist in a single place.

```suggestion
  /// The number of releases to display.
  int get limit {
    final parsedLimit = int.tryParse(results['limit'] as String) ?? defaultLimit;
    return parsedLimit >= 0 ? parsedLimit : defaultLimit;
  }
```